kernel/kernel-api/src/test/scala/io/delta/kernel/internal/types/TypeWideningCheckerSuite.scala

```diff
@@ -200,4 +202,153 @@ class TypeWideningCheckerSuite extends AnyFunSuite {
       new DecimalType(15, 3)))
     assert(!TypeWideningChecker.isIcebergV2Compatible(LongType.LONG, new DecimalType(20, 0)))
   }
+
+  test("assertTableReadable - does nothing if type widening not supported") {
+    val protocol = createProtocolWithoutTypeWidening()
+    val fieldWithTypeChange = createFieldWithTypeChange(
+      "value",
+      LongType.LONG,
+      IntegerType.INTEGER,
+      LongType.LONG)
+    val schema = new StructType(java.util.List.of(fieldWithTypeChange))
+    val metadata = createMetadata(schema)
+
+    TypeWideningChecker.assertTableReadable(protocol, metadata)
+  }
+
+  test("assertTableReadable - does nothing if no type widening metadata") {
+    val protocol = createProtocolWithTypeWidening()
+    val schema = new StructType()
+      .add("id", IntegerType.INTEGER)
+      .add("name", StringType.STRING)
+    val metadata = createMetadata(schema)
+
+    TypeWideningChecker.assertTableReadable(protocol, metadata)
+  }
+
+  test("assertTableReadable - throws on unsupported type change in nested struct") {
+    val protocol = createProtocolWithTypeWidening()
+    val nestedFieldWithTypeChange = createFieldWithTypeChange(
+      "age",
+      IntegerType.INTEGER,
```

Comment:
I'm having trouble reading this, but it feels like this type change should be supported, we probably want to skip the actual nested types for comparison.